software/controller/lib/hal/adc.cpp

```diff
@@ -89,10 +89,20 @@ static constexpr int MaxAdcReading = (OversampleLog2 >= 4) ? 65536 : (1 << (12 +
 // Set sample time ([RM] 16.4.12). I'm using 92.5 A/D clocks to sample.
 // A/D sample time in CPU clock cycles.  Fixed for now
 // This is the time we give the analog input to charge the A/D sampling cap.
-static constexpr int AdcSampleTime = 92;
+enum class ADCSampleTime {
+  s2_5 = 2,
+  s6_5 = 6,
+  s12_5 = 12,
+  s24_5 = 24,
+  s47_5 = 47,
+  s92_5 = 92,
+  s247_5 = 247,
+  s640_5 = 640
+};
+static constexpr ADCSampleTime AdcSampleTime{ADCSampleTime::s92_5};
 
 // Time of an A/D conversion in CPU clock cycles.  This is the sample time + 13
-static constexpr int AdcConversionTime = AdcSampleTime + 13;
+static constexpr int AdcConversionTime{static_cast<uint32_t>(AdcSampleTime) + 13};
```

Comment:
Do we also maybe have some info on why it is 13?

Reply:
yup, tracked it down to 16.4.16, it's actually `SampleTime + 0.5 + 1*resolution`. We haven't defined a resolution constant, since we use the default value (12 bits, which happens to be the highest one). I will define this more clearly.